org.eclipse.dawnsci.hdf5/src/org/eclipse/dawnsci/hdf5/HDF5Utils.java

```diff
@@ -445,39 +446,41 @@ private static Dataset loadDataset(final String fileName, final String node,
 	 * @return dataset
 	 * @throws NexusException
 	 */
-	public static Dataset readDataset(HDF5File f, final String node, final int[] start, final int[] count,
+	public static Dataset readDataset(HDF5File f, final String dataPath, final int[] start, final int[] count,
```

Comment:
![CRITICAL](https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL') Refactor this method to reduce its Cognitive Complexity from 31 to the 15 allowed. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS3776)
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Method has 8 parameters, which is greater than 7 authorized. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS00107)